modules/core/src/lifecycle/component.js

```diff
@@ -32,7 +32,20 @@ export default class Component {
 
   // clone this layer with modified props
   clone(newProps) {
-    return new this.constructor(Object.assign({}, this.props, newProps));
+    const {props} = this;
+
+    // Async props cannot be copied with Object.assign. Extract the plain value.
```

Comment:
The last part of the comment doesn't seem helpful...

Extract the plain value. => Copy them separately?